### PR TITLE
ENH: improve docstring of correlate and correlate2d to fix #1968

### LINKED ISSUE #1968: correlate2d's output does not agree with correlate's output in scipy.signal module (Trac #1443)
_Original ticket http://projects.scipy.org/scipy/ticket/1443 on 2011-05-23 by trac user SevenSea, assigned to unknown._

When the 2-dimensional array size is even, correlate2d's output does not agree with correlate's output in scipy.signal module. There is a shift of index in the output of correlate2d function. (Information for comparition: convolv2d works good.)

Testing code:

```
import numpy as np
import scipy.signal as sn

d = np.ones((4,4))
print "data: "
print d

print "correlate(d,d,'same'): "
print sn.correlate(d,d,'same')

print "correlate2d(d,d,'same'): "
print sn.correlate2d(d,d,'same')
```

The output:

```
data: 
[[ 1.  1.  1.  1.]
 [ 1.  1.  1.  1.]
 [ 1.  1.  1.  1.]
 [ 1.  1.  1.  1.]]
correlate(d,d,'same'): 
[[  4.   6.   8.   6.]
 [  6.   9.  12.   9.]
 [  8.  12.  16.  12.]
 [  6.   9.  12.   9.]]
correlate2d(d,d,'same'): 
[[  9.  12.   9.   6.]
 [ 12.  16.  12.   8.]
 [  9.  12.   9.   6.]
 [  6.   8.   6.   4.]]
```

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
To fix #1968

#### What does this implement/fix?
To improve docstring of correlate and correlate2d

### CODE REVIEW

File: scipy/signal/signaltools.py [hunk 1]
```diff
@@ -166,6 +166,8 @@ def correlate(in1, in2, mode='full', method='auto'):
     `fftconvolve`. In certain cases (i.e., arrays of objects or when
     rounding integers can lose precision), ``method='direct'`` is always used.
 
+    Outputs of `correlate` and `correlate2d` are different when you are using the "same" mode and even-length inputs.
```
Comment (ilayn) [inline scipy/signal/signaltools.py]: PEP8; longer than 80 characters

File: scipy/signal/signaltools.py [hunk 2]
```diff
@@ -1593,6 +1595,10 @@ def correlate2d(in1, in2, mode='full', boundary='fill', fillvalue=0):
         A 2-dimensional array containing a subset of the discrete linear
         cross-correlation of `in1` with `in2`.
 
+    Notes
+    -----
+    Outputs of `correlate` and `correlate2d` are different when you are using the "same" mode and even-length inputs.
```
Comment (ilayn) [inline scipy/signal/signaltools.py]: PEP8; longer than 80 characters

File: scipy/signal/signaltools.py [hunk 3]
```diff
@@ -1593,6 +1595,10 @@ def correlate2d(in1, in2, mode='full', boundary='fill', fillvalue=0):
         A 2-dimensional array containing a subset of the discrete linear
         cross-correlation of `in1` with `in2`.
 
+    Notes
```
Comment (ilayn) [inline scipy/signal/signaltools.py]: Notes section is usually before the examples for a better readability.

**Review by ilayn — CHANGES_REQUESTED** [review]
Hi @AtsushiSakai some minor comments.

File: scipy/signal/signaltools.py (hunk 3 above)
Comment (AtsushiSakai) [reply scipy/signal/signaltools.py]: I think already the Notes section is "before" the examples. You mean "after"?

File: scipy/signal/signaltools.py (hunk 1 above)
Comment (AtsushiSakai) [reply scipy/signal/signaltools.py]: Thanks I fixed it.

File: scipy/signal/signaltools.py (hunk 2 above)
Comment (AtsushiSakai) [reply scipy/signal/signaltools.py]: Thanks I fixed it.

File: scipy/signal/signaltools.py (hunk 3 above)
Comment (ilayn) [reply scipy/signal/signaltools.py]: Ah sorry, my bad. Got confused with the previous part. Nevermind

**Comment by endolith** [comment]
That's fine, but I would give a little more detail, since the only difference is an offset by one sample.

**Comment by endolith** [comment]
> When using "same" mode with even-length inputs, the outputs of `correlate` and `correlate2d` differ: There is a 1-index offset between them.

or something like that

**Comment by AtsushiSakai** [comment]
@endolith Thank you. I fixed it.

**Review by endolith — APPROVED** [review]

**Comment by AtsushiSakai** [comment]
Hi. @ilayn  I fixed these based on your review. Please take a look.

**Comment by ilayn** [comment]
Thanks @AtsushiSakai !

**Pull request merged**